src/runner/process-batch.ts

```diff
@@ -216,7 +218,7 @@ export async function processBatchAsync(
           }
         }
 
-        await fs.outputFile(tsFilePath, newFileText);
+        await fs.outputFile(tsFilePath, newFileText, { mode: stats.mode });
```

Comment:
What is this doing? Preserving the +x bit?

Reply:
That's correct.  I'll add a comment about this.